smoke/LeafDevice/details/Details.cs

```diff
@@ -148,8 +163,9 @@ protected async Task GetOrCreateDeviceIdentityAsync()
             IotHubConnectionStringBuilder builder = IotHubConnectionStringBuilder.Create(this.iothubConnectionString);
             RegistryManager rm = RegistryManager.CreateFromConnectionString(builder.ToString());
 
-            var edgeScope = Option.None<string>();
-            this.edgeDeviceId.ForEach(async (id) => edgeScope = await GetScopeIfExitsAsync(rm, id));
+            Option<string> edgeScope = await this.edgeDeviceId.Map(async (id) => {
+                return await GetScopeIfExitsAsync(rm, id);
+            }).GetOrElse(() => Task.FromResult<Option<string>>(Option.None<string>()));
```

Comment:
You don't need the `async`/`await` here. 

```csharp
Option<string> edgeScope = await this.edgeDeviceId
    .Map(id => GetScopeIfExistsAsync(rm, id))
    .GetOrElse(() => Task.FromResult<Option<string>>(Option.None<string>()));
```

Reply:
Thanks fixed.